PhysicsTools/NanoAOD/python/electrons_cff.py

```diff
@@ -438,8 +469,11 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
 electronSequence = cms.Sequence(bitmapVIDForEle + isoForEle + ptRatioRelForEle + slimmedElectronsWithUserData + finalElectrons)
 electronTables = cms.Sequence (electronMVATTH + electronTable)
 electronMC = cms.Sequence(electronsMCMatchForTable + electronMCTable)
+from RecoEgamma.ElectronIdentification.heepIdVarValueMapProducer_cfi import heepIDVarValueMaps
+_updateTo106X_sequence =cms.Sequence(heepIDVarValueMaps + slimmedElectronsTo106X)
+heepIDVarValueMaps.dataFormat = 2
```

Comment:
it looks like comments from the review a week ago do not have any follow up.
@Sam-Harper please check.
Thank you.

Reply:
yes sorry I missed them as I was sort of away last week. 

The implication that nano will never run on AOD inputs, miniAOD has to be present, even if its being produced on the fly in the job.  Therefore we force it to always use the miniAOD inputs, even if AOD is present, it is more efficient this way as AOD it has to approximate the packed candidates in a special sequence to avoid rounding errors going between AOD and miniAOD 

So =2 is the correct behaviour here as we always want to run on miniAOD

Reply:
the point here is that it will not work with miniAOD on the fly because  heepIDVarValueMaps is apparently used to create some IDs in the miniAOD itself and you are modifying it here in a nano configuration, which is a wrong thing to do in the context of possibly running together with miniAOD.
So, if a miniAOD part can possibly be considered to be made on the fly together with NANO modules, this change should be made in a different way.

Reply:
Yes, I see your point. In fact I now realise as a result nano can not run at the same time as miniAOD due to how it uses VID.  So this doesnt really do anything it always needs miniAOD and AOD is not an option.  Can I just leave it?

Reply:
I take note of this, we can address it when preparing a merged MINI+NANO workflow that we don't need at the moment, though

Reply:
OK

Reply:
thank you. Also @peruzzim , when you are doing that let me know, we might have to revisit a few things (its not hard to do it inefficiently, just by having two seperate VID modules but perhaps not optimal)